tests/specs/benchmarks/verification.k

```diff
@@ -281,75 +236,59 @@ module VERIFICATION
+    rule         #fii(_,_,_,_,_) <Int pow256 => true [simplification]
 
-// definition of #signer
+    // axiomatization of #isValid
 
-rule #signer(SIGS, I,  DATA_HASH) => #symEcrec(#ecrecData(SIGS, I, DATA_HASH)) requires I >=Int 0 andBool   #sigV(SIGS, I) =/=Int 0 andBool #sigV(SIGS, I) =/=Int 1   andBool notBool #ecrecEmpty(#ecrecData(SIGS, I, DATA_HASH))
-rule #signer(SIGS, I,  DATA_HASH) => 0                                         requires I >=Int 0 andBool   #sigV(SIGS, I) =/=Int 0 andBool #sigV(SIGS, I) =/=Int 1   andBool         #ecrecEmpty(#ecrecData(SIGS, I, DATA_HASH))
-rule #signer(SIGS, I, _DATA_HASH) => #sigR(SIGS, I)                            requires I >=Int 0 andBool ( #sigV(SIGS, I)  ==Int 0  orBool #sigV(SIGS, I)  ==Int 1 )
+    rule #isValid(SIGS, I, DATA_HASH, DATA_BUF, STORAGE, MSG_SENDER, _) => true   requires I  <Int #fii(SIGS, DATA_HASH, DATA_BUF, STORAGE, MSG_SENDER)
+    rule #isValid(SIGS, I, DATA_HASH, DATA_BUF, STORAGE, MSG_SENDER, _) => false  requires I ==Int #fii(SIGS, DATA_HASH, DATA_BUF, STORAGE, MSG_SENDER)
 
-syntax ByteArray ::= #ecrecData ( ByteArray , Int , Int ) [function]
-rule #ecrecData(SIGS, I, DATA_HASH) => #encodeArgs(#bytes32(DATA_HASH), #uint8(#sigV(SIGS,I)), #bytes32(#sigR(SIGS,I)), #bytes32(#sigS(SIGS,I)))
+    // axiomatization of #isValidSig for each signature type
 
-  // ########################
-  // Account Map
-  // ########################
+    rule #isValidSig(SIGS, I, DATA_HASH, _DATA_BUF, _STORAGE, _MSG_SENDER) => notBool #ecrecEmpty(#ecrecData(SIGS, I, DATA_HASH))
+      requires #sigV(SIGS, I) =/=Int 0
+       andBool #sigV(SIGS, I) =/=Int 1
 
-    syntax Int ::= "#CONTRACT_ID"   [function]
+    // definition of #signer
 
-    rule 0 <=Int #CONTRACT_ID               => true
-      [simplification]
-    rule         #CONTRACT_ID   <Int pow160 => true
-      [simplification]
-    rule         #CONTRACT_ID   <Int pow256 => true
-      [simplification]
-
-    syntax Int ::= "#CALLEE_ID"   [function]
-
-    rule 0 <=Int #CALLEE_ID               => true
-      [simplification]
-    rule         #CALLEE_ID   <Int pow160 => true
-      [simplification]
-    rule         #CALLEE_ID   <Int pow256 => true
-      [simplification]
+    rule #signer(_   , I, _         ) => 0                                          requires I  <Int 0 // address lastOwner = address(0);
+    rule #signer(SIGS, I,  DATA_HASH) => 0                                          requires I >=Int 0 andBool   #sigV(SIGS, I) =/=Int 0 andBool #sigV(SIGS, I) =/=Int 1   andBool         #ecrecEmpty(#ecrecData(SIGS, I, DATA_HASH))
+    rule #signer(SIGS, I,  DATA_HASH) => #symEcrec(#ecrecData(SIGS, I, DATA_HASH))  requires I >=Int 0 andBool   #sigV(SIGS, I) =/=Int 0 andBool #sigV(SIGS, I) =/=Int 1   andBool notBool #ecrecEmpty(#ecrecData(SIGS, I, DATA_HASH))
+    rule #signer(SIGS, I, _DATA_HASH) => #sigR(SIGS, I)                             requires I >=Int 0 andBool ( #sigV(SIGS, I)  ==Int 0  orBool #sigV(SIGS, I)  ==Int 1 )
```

Comment:
AFAICT, none of these are used in any of the proofs. Can we just remove this block?

Reply:
Ah wait, I see uses of `#ecrec` below...